### DISCUSSION TITLE
Use of dropout for node level prediction

### DISCUSSION THREAD

**Discussion by kimzed** [pyg-team/pytorch_geometric#4705] [Q&A]
Currently I am using dropout in my model as I have seen that it is quite standard for GNN architectures. I am trying to fine tune my model (which already has good results). I do not really grasp well what could be to drop certain node features for node prediction. Doesn't it make prediction more difficult, without any benefit? I am wondering whether I am using a trick that would be actually more fitted for graph level prediction.

To put some context, here is the forward function from my model:

``` python
data = self.initial_layer(data)
data.x = torch.tanh(data.x)
data.x = F.dropout(data.x, training=self.training, p=0.3)
data = self.layer1(data)
data.x = torch.tanh(data.x)
data.x = F.dropout(data.x, training=self.training, p=0.3)
data = self.layer2(data)
data.x = torch.tanh(data.x)
data.x = F.dropout(data.x, training=self.training, p=0.3)
data.x = self.out(data.x)
```

**Comment by rusty1s** [comment]
The goal of applying dropout is to avoid overfitting. In that way, you force the model to make correct predictions given only a subset of data available. If your model already performs well, and there is no significant overfitting taking place, you do not need to necessarily apply dropout.
